src/components/implementation/no_interface/scheddev/fprr/sl_mod_fprr.c

```diff
@@ -70,10 +74,32 @@ sl_mod_thd_delete(struct sl_thd_policy *t)
 void
 sl_mod_thd_param_set(struct sl_thd_policy *t, sched_param_type_t type, unsigned int v)
 {
-	assert(type == SCHEDP_PRIO && v < SL_FPRR_NPRIOS);
-	ps_list_rem_d(t); 	/* if we're already on a list, and we're updating priority */
-	t->priority = v;
-	ps_list_head_append_d(&threads[t->priority], t);
+	switch (type) {
+	case SCHEDP_PRIO:
+	{
+		assert(v < SL_FPRR_NPRIOS);
+		ps_list_rem_d(t); 	/* if we're already on a list, and we're updating priority */
+		t->priority = v;
+		ps_list_head_append_d(&threads[t->priority], t);
+		sl_thd_setprio(sl_mod_thd_get(t), t->priority);
+
+		break;
+	}
+	case SCHEDP_WINDOW:
+	{
+		struct sl_thd *td = sl_mod_thd_get(t);
+
+		assert(v >= SL_FPRR_US_MIN && v <= SL_FPRR_US_MAX);
```

Comment:
My bad, period doesn't need to have that check. It was mainly for budget as budget is `unsigned long`, I will get rid of this for Period.